src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -707,6 +716,21 @@ private void UpdateNodeStatus(BuildEventContext buildEventContext, NodeStatus? n
     /// </summary>
     private void TargetFinished(object sender, TargetFinishedEventArgs e)
     {
+        var buildEventContext = e.BuildEventContext;
+        if (_restoreContext is null
+            && buildEventContext is not null
+            && e.TargetName == "GetTargetPath"
+            && _projects.TryGetValue(new ProjectContext(buildEventContext), out Project? project))
```

Comment:
Running these checks on every TargetFinished feels a bit expensive to me even though it's mostly just a string comparison. Do you think it would make sense to store a bool `_hasUsedCache` and check that early to make it even cheaper?

Reply:
Yes, that makes sense to do, thanks for the suggestion!